server/container_create_linux.go

```diff
@@ -202,7 +204,19 @@ func (s *Server) addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container,
 			return nil, nil, errors.New("mount.ContainerPath is empty")
 		}
 
-		if m.Image != nil && m.Image.Image != "" {
+		if m.GetImage().GetImage() != "" {
+			// Try mountArtifact first, and fall back to mountImage if it fails with ErrNotFound
+			artifactVolumes, err := s.mountArtifact(ctx, specgen, m)
+			if err == nil {
+				volumes = append(volumes, artifactVolumes...)
+
+				continue
+			}
+
+			if !errors.Is(err, ociartifact.ErrNotFound) {
+				return nil, nil, fmt.Errorf("mount artifact: %w", err)
```

Comment:
I'm not sure, should we still try to mount an image if `!errors.Is(err, ociartifact.ErrNotFound)`?

Reply:
fair enough. thanks!